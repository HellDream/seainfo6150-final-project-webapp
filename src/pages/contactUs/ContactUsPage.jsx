import React, {useEffect, useState} from 'react';
import styles from './ContactUsPage.module.css';
import Header from '../../components/headers/Header.jsx';
import Dialog from '../../components/dialog/Dialog';

const ContactUsPage = (props) => {
    const [dialogState, setDialogState] = useState({
        open: false,
        message: '',
        ok: null,
        cancel: null,
        onClose: null,
    });
    useEffect(()=>{
        document.title = "Thrift Store - Contact Us";

    });

    const handleSubmit = (event) =>{
        event.preventDefault();
        setDialogState({
            open: true,
            message: 'You have uploaded your item successfully!',
            ok: () => {
                setDialogState({ ...dialogState, open: false });
                props.history.goBack();
            },
            onClose: () => {
                setDialogState({ ...dialogState, open: false });
            },
        });  
    }
    return (
        <>
            <Dialog
                open={dialogState.open}
                message={dialogState.message}
                ok={dialogState.ok}
                onClose={dialogState.onClose}
            />
            <Header
                createPage={true}
                onCancel={() => {
                    props.history.goBack();
                }}
            />
            <section className={styles.section}>
                <div className={styles.container}>
                    <h3 className={styles.formTitle}>
                        Tell Us What You Think!
                    </h3>
                    <form className={styles.form} onSubmit={handleSubmit}>
                        <div className={styles.inputContainer}>
                            <label  className={styles.label} htmlFor="message">Your Email:</label>
                            <input type="email" className={styles.input} required />
                        </div>
                        <div className={styles.inputContainer}>
                            <label className={styles.label} htmlFor="message">
                                Your Message:
                            </label>
                            <textarea
                                className={styles.inputTextArea}
                                type="text"
                                id="message"
                                name="message"
                                required
                            />
                        </div>
                        <button className={styles.submitButton} type="submit">
                            Submit
                        </button>
                    </form>
                </div>
            </section>
        </>
    );
};

export default ContactUsPage;
