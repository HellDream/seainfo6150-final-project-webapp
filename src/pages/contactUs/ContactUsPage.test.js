import React from 'react';
import ContactUsPage from './ContactUsPage';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

describe('ContactUsPage tests', () => {
    it('renders correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <ContactUsPage />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
