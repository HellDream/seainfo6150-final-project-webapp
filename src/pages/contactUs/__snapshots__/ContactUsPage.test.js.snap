// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUsPage tests renders correctly 1`] = `
<div>
  <div />
  <div>
    <div />
    <header
      class="header"
    >
      <div
        class="navbar"
      >
        <div
          class="empty"
        />
        <a
          class="homeLink"
          href="/"
        >
          <h3
            class="title"
          >
            thrift store
          </h3>
        </a>
        <div
          class="links"
        >
          <button
            class="cancel"
          >
            cancel
          </button>
        </div>
      </div>
    </header>
  </div>
  <section
    class="section"
  >
    <div
      class="container"
    >
      <h3
        class="formTitle"
      >
        Tell Us What You Think!
      </h3>
      <form
        class="form"
      >
        <div
          class="inputContainer"
        >
          <label
            class="label"
            for="message"
          >
            Your Email:
          </label>
          <input
            class="input"
            required=""
            type="email"
          />
        </div>
        <div
          class="inputContainer"
        >
          <label
            class="label"
            for="message"
          >
            Your Message:
          </label>
          <textarea
            class="inputTextArea"
            id="message"
            name="message"
            required=""
            type="text"
          />
        </div>
        <button
          class="submitButton"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </section>
</div>
`;
